### PR TITLE
feat(mae-consumer): enable mae-consumer to use ssl when communicating…

### PR DESCRIPTION
**Pull request by thomasplarsson**
… with elasticsearch

Fixes: #2129

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.linkedin.metadata.utils.elasticsearch;
+package com.linkedin.metadata.kafka.elasticsearch;
```
Comment (dexter-mh-lee) [inline metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java:1]: Is this class getting referenced anywhere? Why do we need an MCE related class in the MAE consumer?

**Review by dexter-mh-lee — COMMENTED** [review]
Overall love the clean-up. 

Have you tested whether this works for gms? Seems like the search dao factories like https://github.com/linkedin/datahub/blob/master/gms/factories/src/main/java/com/linkedin/gms/factory/dataset/DatasetSearchDAOFactory.java is using the client.

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java (hunk 1 above)
Comment (jplaisted) [reply metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java:1]: I agree it seems misnamed, but it looks used / needed.

`ElasticEvent` (the interface) is referenced in more places. This is just a concrete implementation (the only one?), and is created here.

https://github.com/linkedin/datahub/blob/f103998dcb1ade1a0b21fce4b1cbcb6bb17a37e9/metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/MetadataAuditEventsProcessor.java#L112

`ElaticEvent` looks like it contains basic information about an update, like which index it should go to, and the id of the document. The `buildJson` method is abstract. So the MAE job will take the MAE, pull off the aspects, transform them to search documents, and create an `ElaticEvent` for each that determines the ID and index to send the document to.

Comment (dexter-mh-lee) [reply metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java:1]: Got it. Should we rename it to MAEElasticEvent?

**Comment by dexter-mh-lee** [comment]
> Have you tested whether this works for gms? Seems like the search dao factories like https://github.com/linkedin/datahub/blob/master/gms/factories/src/main/java/com/linkedin/gms/factory/dataset/DatasetSearchDAOFactory.java is using the client.

My bad. The client is in a central place. Just the connector config. LGTM!

**Review by dexter-mh-lee — APPROVED** [review]
Confirmed that this change works on my local env.

File: metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java (hunk 1 above)
Comment (shirshanka) [reply metadata-jobs/mae-consumer-job/src/main/java/com/linkedin/metadata/kafka/elasticsearch/MCEElasticEvent.java:1]: Good ideas. I would suggest doing that refactoring as a separate PR to keep this one contained.

**Review by shirshanka — APPROVED** [review]
LGTM.

**Pull request merged**